packages/shadydom/src/patch-events.js

```diff
@@ -391,7 +394,14 @@ function targetNeedsPathCheck(node) {
 /** @this {Node} */
 export function dispatchEvent(event) {
   flush();
-  return this[utils.NATIVE_PREFIX + 'dispatchEvent'](event);
+  if (this instanceof Node && !document.documentElement[utils.NATIVE_PREFIX + 'contains'](this)) {
```

Comment:
The `this instanceof Node` check here is not something I tried in the perf tests: I hadn't considered non-Node event targets and Node's `contains` throws if you pass it a non-Node.

Reply:
We talked about this in a meeting and decided `this instanceof Node` shouldn't be a perf problem and there's another thread about `document.contains` vs `document.documentElement.contains` below so I'm going to close this one.